files/scripts/mark_dhcp_packet.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+import os
+import subprocess
+import sys
+import time
+
+from sonic_py_common import logger
+from swsscommon import swsscommon
+
+log = logger.Logger('mark_dhcp_packet')
+
+class MarkDhcpPacket(object):
+    """
+    Class used to configure dhcp packet mark in ebtables
+    """
+
+    def __init__(self):
+        self.config_db_connector = None
+        self.state_db_connector = None
+
+    @property
+    def config_db(self):
+        """
+        Returns config DB connector.
+        Initializes the connector during the first call
+        """
+        if self.config_db_connector is None:
+            self.config_db_connector = swsscommon.ConfigDBConnector()
+            self.config_db_connector.connect()
+
+        return self.config_db_connector
+
+    @property
+    def state_db(self):
+        """
+        Returns the state DB connector.
+        Initializes the connector during the first call
+        """
+        if self.state_db_connector is None:
+            self.state_db_connector = swsscommon.SonicV2Connector(host='127.0.0.1')
+            self.state_db_connector.connect(self.state_db_connector.STATE_DB)
+
+        return self.state_db_connector
+
+    @property
+    def is_dualtor(self):
+        """
+        Checks if script is running on a dual ToR system
+        """
+        localhost_key = self.config_db.get_keys('DEVICE_METADATA')[0]
+        metadata = self.config_db.get_entry('DEVICE_METADATA', localhost_key)
+
+        return 'subtype' in metadata and 'dualtor' in metadata['subtype'].lower()
+
+    def get_mux_intfs(self):
+        """
+        Returns a list of mux cable interfaces
+        """
+        mux_cables = self.config_db.get_table('MUX_CABLE')
+        mux_intfs = [intf for intf in mux_cables if intf.startswith('Ethernet')]
+
+        return mux_intfs
+
+    def generate_mark_for_intf(self, intf):
+        '''
+        type: string, format: hexadecimal
+        '''
+        intf_index_str = intf[8:]
```

Comment:
Please specify an example here. Another issue is, if we ever change the name to say etp1, then this logic fails. So please revise the logic to uniquely identify the interface index